src/reports/reporter.ts

```diff
@@ -11,24 +13,37 @@ export enum EventName {
   START_BINARY = "tabnine-binary-run",
 }
 
+let reporter: TelemetryReporter;
+
 export function initReporter(
   context: ExtensionContext,
   id: string,
   version: string,
   key: string
 ): void {
+  if (inTestMode) return;
+
   reporter = new TelemetryReporter(id, version, key);
   context.subscriptions.push(reporter);
 }
 
 export function report(event: EventName): void {
-  reporter.sendTelemetryEvent(event);
+  if (inTestMode) return;
+
+  void (async () => {
```

Comment:
why the anonymous function ? why not  change the `report` function to be promise aware?

Reply:
I think it makes the api cleaner this way, so you don't need to 'color' all caller functions / `void` it